external/fv3fit/tests/test_models.py

```diff
@@ -76,3 +77,35 @@ def test_nonnegative_model_outputs():
     model.fit([batch])
     prediction = model.predict(batch)
     assert prediction.min() >= 0.0
+
+
+def test_DenseModel_clipped_inputs():
+    hyperparameters = DenseHyperparameters(
+        ["a", "b"],
+        ["c"],
+        packer_config=PackerConfig({"a": {"z": SliceConfig(None, 3)}}),
+    )
+    model = DenseModel(["a", "b"], ["c"], hyperparameters,)
+
+    nz = 5
+    dims = ["x", "y", "z"]
+    shape = (2, 2, nz)
+    arr = np.arange(np.prod(shape)).reshape(shape).astype(float)
+    input_data = xr.Dataset({"a": (dims, arr), "b": (dims, arr), "c": (dims, arr + 1)})
+    model.fit([input_data])
+    model.predict(input_data)
+    assert model.X_packer._n_features["a"] == 3
+    assert model.X_packer._n_features["b"] == 5
```

Comment:
This assertion makes a lot of assumptions about the internal workings of the model, and I'm worried it could continue to pass despite the model being broken. Can you add a test that asserts that a model's predictions are unchanged if the values it is supposedly clipping are set to NaN in the inputs?